src/Adapter/DbTableGateway.php

```diff
@@ -11,20 +11,20 @@
 
 use Zend\Db\Sql\Where;
 use Zend\Db\Sql\Having;
-use Zend\Db\TableGateway\TableGateway;
+use Zend\Db\TableGateway\AbstractTableGateway;
 
 class DbTableGateway extends DbSelect
 {
     /**
      * Constructs instance.
      *
-     * @param TableGateway                      $tableGateway
+     * @param AbstractTableGateway              $tableGateway
      * @param null|Where|\Closure|string|array  $where
      * @param null|string|array                 $order
      * @param null|string|array                 $group
      * @param null|Having|\Closure|string|array $having
      */
-    public function __construct(TableGateway $tableGateway, $where = null, $order = null, $group = null, $having = null)
+    public function __construct(AbstractTableGateway $tableGateway, $where = null, $order = null, $group = null, $having = null)
```

Comment:
I was going to ask why you were type-hinting on `AbstractTableGateway` and not `TableGatewayInterface`, and then noticed that we call the `getSql()` method, which is only available in the abstract implementation...